### ISSUE TITLE
No CUDA GPUs are available

### ISSUE THREAD

**Issue by Douuble** [cvg/limap#61]
We have already set up the whole environment of COLMAP and LIMAP (gui window pops up after running _colmap gui_, and there is no error after running _python3 -c "import limap_"). But when we tried to run the Quickstart command _python runners/hypersim/triangulation.py --output_dir outputs/quickstart_triangulation_ on the Ubuntu virtual environment in the  VMware locally, it failed with the error I attached. I wonder if there is the way that can we run it without cuda and with just CPUs?
<img width="362" alt="error" src="https://github.com/cvg/limap/assets/101459673/841e7fa8-a2e1-482c-a681-846c883d79cc">

**Comment by B1ueber2y** [comment]
Hey. Thanks for the interest!

The feature extraction and matching for lines are mostly benefited from deep learning, thus cannot run on CPU. Nonethess, the classic detector and matcher LSD + LBD is applicable when GPU is not available, though leading to degraded performance and efficiency.

I have just supported a CPU-friendly options in the configurations. To be able to run on devices without CUDA GPU, you can run:

```
python runners/hypersim/fitnmerge --default_config_file cfgs/fitnmerge/default_cpu.yaml --output_dir outputs/quickstart_fitnmerge
```

Same also applies to triangulation where the default config file will be ``cfgs/triangulation/default_cpu.yaml``

Thanks again!

**Comment by Douuble** [comment]
Thank you for your reply and improvement! !

I tried the modified version, but I think maybe you forget the `.py`, I run:
`python runners/hypersim/fitnmerge.py --default_config_file cfgs/fitnmerge/default_cpu.yaml --output_dir outputs/quickstart_fitnmerge`.

And I encountered the new error listed below:
![image](https://github.com/cvg/limap/assets/101459673/36fa1b9c-279b-4974-a347-e81297aef703)
I tried some solutions like reducing the version of joblib, but it didn't work. I wonder if you have any ideas about it.
